chain/consensus/mir/crypto_test.go

```diff
@@ -48,16 +51,43 @@ func (n *cryptoNode) WalletSign(ctx context.Context, k address.Address, msg []by
 	signature, err = n.w.WalletSign(ctx, k, msg, msgMeta)
 	return
 }
-func (n *cryptoNode) WalletVerify(ctx context.Context, k address.Address, msg []byte, sig *filcrypto.Signature) (bool, error) {
+func (n *cryptoWallet) WalletVerify(ctx context.Context, k address.Address, msg []byte, sig *filcrypto.Signature) (bool, error) {
 	err := sigs.Verify(sig, k, msg)
 	return err == nil, err
 }
 
+// ---------------
+
+type cryptoNode struct {
+	privKey libp2pcrypto.PrivKey
+	id      peer.ID
+}
+
+func newCryptoNode() (*cryptoNode, error) {
```

Comment:
So we are using a new identity for Mir instead of using the one from our `libp2p.Host` right? I guess this is OK for now, we can always change it once we integrate the transport layer.

Reply:
We will use the real node ID retrieved via `host.ID()`. `newCryptoNode` is a dummy implementation of the interface to not inject the real full node API in unit tests.